File: frontend/src/types/property.ts
export type PropertyType =
  | "plot"
  | "house"
  | "apartment"
  | "commercial"
  | "villa";

export type ListingType = "sale" | "rent";

export type FurnishingStatus = "unfurnished" | "semi_furnished" | "fully_furnished";

export type PropertySource =
  | "geb"
  | "external"
  | "broker";


export interface Property {
  id: string;
  title: string;
  propertyType: PropertyType;
  listingType?: ListingType;
  price: number;
  monthlyRent?: number;
  securityDeposit?: number;
  availableFrom?: string;
  furnishingStatus?: FurnishingStatus;
  bhk?: number;

  area: number;
  areaUnit: "sqft" | "sqm";
  city: string;
  locality: string;
  latitude: number;
  longitude: number;
  image: string;
  images?: string[];
  source: PropertySource;
  sourceName: string;
  sourceUrl?: string;
  investmentScore?: number;
  featured?: boolean;
  status?: string;
  rejectionReason?: string;
  description?: string;
  sellerId?: string;

  seller?: {
    id: string;
    full_name: string;
    email: string;
    phone?: string;
    avatar_url?: string;
  };

  createdAt?: string;
  updatedAt?: string;
}

File: backend/app/schemas/property.py
from typing import Optional
from pydantic import BaseModel, Field


class PropertyCreate(BaseModel):
    title: str = Field(
        min_length=3,
        max_length=200,
    )
    property_type: str
    price: float = Field(gt=0)
    listing_type: str = "sale"  # "sale" | "rent"
    monthly_rent: Optional[float] = None
    security_deposit: Optional[float] = None
    available_from: Optional[str] = None
    furnishing_status: Optional[str] = None  # "unfurnished" | "semi_furnished" | "fully_furnished"
    bhk: Optional[int] = None
    area: Optional[float] = Field(
        default=None,
        gt=0,
    )
    area_unit: str = "sqft"
    city: str
    locality: Optional[str] = None
    latitude: float
    longitude: float
    image: Optional[str] = None
    images: list[str] = Field(
        default_factory=list
    )
    description: Optional[str] = None


class PropertyResponse(PropertyCreate):
    id: str
    source: str
    source_name: Optional[str] = None
    source_url: Optional[str] = None
    investment_score: Optional[float] = None
    featured: bool
    status: str
    seller_id: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
